connector-builder-server/src/main/openapi/openapi.yaml

```diff
@@ -0,0 +1,217 @@
+  title: Connector Builder Server API
+  contact:
+    email: contact@airbyte.io
+  license:
+    name: MIT
+    url: "https://opensource.org/licenses/MIT"
+externalDocs:
+  description: Find out more about Connector Builder
+  url: "https://docs.airbyte.com/connector-development/config-based/overview/"
+
+paths:
+  /v1/stream/read:
+    post:
+      summary: Reads a specific stream in the source
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/StreamReadRequestBody"
+        required: true
+      responses:
+        "200":
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/StreamRead"
+        "400":
+          $ref: "#/components/responses/ExceptionResponse"
+        "422":
+          $ref: "#/components/responses/InvalidInputResponse"
+  /v1/streams/list:
+    post:
+      summary: List all streams present in the connector definition, along with their specific request URLs
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/StreamsListRequestBody"
+        required: true
+      responses:
+        "200":
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/StreamsListRead"
+        "400":
+          $ref: "#/components/responses/ExceptionResponse"
+        "422":
+          $ref: "#/components/responses/InvalidInputResponse"
+
+components:
+  schemas:
+    StreamRead:
+      type: object
+      required:
+        - results
+      properties:
+        results:
+          type: array
+          description: The RECORD and STATE AirbyteMessages coming from the read operation
+          items:
+            type: string
+        logs:
+          type: array
+          description: The LOG AirbyteMessages coming from the read operation
+          items:
+            type: string
+        httpRequests:
+          type: array
+          description: HTTP requests sent to the source API, and their responses
+          items:
+            type: object
+            required:
+              - request
+              - response
+            properties:
+              request:
+                type: object
+                description: The HTTP request sent to the source API
+                required:
+                  - url
+                  - headers
+                properties:
+                  url:
+                    type: string
+                    description: URL that the request was sent to
+                  body:
+                    type: object
+                    description: The body of the HTTP request, if present
+                  headers:
+                    type: object
+                    description: The headers of the HTTP request
```

Comment:
I wasn't super sure if these should be JSON objects or just strings. I went with JSON objects here because it looks like that's what they are in the debug log messages of the `read` command, but open to feedback here